tests/vanilla/store.test.tsx

```diff
@@ -579,3 +579,27 @@ it('should update derived atom even if dependances changed (#2697)', () => {
   store.set(primitiveAtom, 1)
   expect(onChangeDerived).toHaveBeenCalledTimes(1)
 })
+
+it('should call subscribers after setSelf updates atom value', () => {
+  const store = createStore()
+  const a = atom(0)
+  let unmount
+  a.onMount = vi.fn(((setSelf) => {
```

Comment:
Can we avoid calling `setSelf`? It wrongly reminds me of `setSelf` in the read function option.

```suggestion
  a.onMount = vi.fn(((set) => {
```

Reply:
I renamed to setAtom.

Reply:
Thanks!